商品削除機能2

naoya1215 opened the pull request:
#What
商品削除機能

#Why
商品削除機能の実装

#提出動画
https://gyazo.com/37669b90c444a0ecb311fc8045ddd770

naoya1215 committed:
商品削除機能実装

naoya1215 committed:
コメント追加

OhyaYouki commented on app/controllers/items_controller.rb:
```diff
@@ -48,6 +48,12 @@ def update # 更新処理
 
+  def destroy #出品商品の削除
+    item = Item.find(params[:id])
+    item.destroy
+    redirect_to root_path
```
```ruby
item.destroy
```
出品者でなければ商品が削除できないように条件分岐を追加しましょう！

#### 理由：
既にビューの方で削除ボタンを表示しないよう対策済みですが、
検証ツールなどを用いることで削除ボタンを押さずともdestroyアクションを動かすことが可能です。
そういった不正な操作への対策としてコントローラー側でも条件分岐が必要となります。

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます。

コメントいたしましたので、ご確認をお願いいたします。

コードの修正後、コミット・プッシュを行った上で再度レビュー依頼をお送りください。
ご対応のほど、よろしくお願いいたします。

naoya1215 committed:
商品出品者のみが削除可能

メンターアドバイス
出品者でなければ商品が削除できないように条件分岐を追加しましょう！

理由：

既にビューの方で削除ボタンを表示しないよう対策済みですが、
検証ツールなどを用いることで削除ボタンを押さずともdestroyアクションを動かすことが可能です。
そういった不正な操作への対策としてコントローラー側でも条件分岐が必要となります。

naoya1215 commented:
コメント有難うございます。
投稿者のみ削除出来るようにif文を用いて修正をしております。
確認をお願いします。

OhyaYouki commented on app/controllers/items_controller.rb:
```diff
@@ -48,6 +48,14 @@ def update # 更新処理
+    # @item = Item.find(params[:id]) #リファクタリング
+    if current_user.id == @item.user_id
+      @item.destroy
+      redirect_to root_path
```
```
redirect_to root_path
```
こちらの記述をifの外に記述しましょう。

理由：
現在の状態ですとifの条件がfalseであった場合`redirect_to root_path`が読まれません。
destroyアクションが動いた後、遷移するviewが存在せず、意図通りの挙動にならないためです。

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます。

修正点確認させていただきました。
細かい点になるのですが、追加でコメントいたしましたので、ご確認いただけますと幸いです。

修正後、再度レビュー依頼フォームをご提出ください。

ご対応のほど、よろしくお願いいたします。

naoya1215 committed:
redirect_to root_pathの記述をifの外へ

メンターコメントにより修正、自動整形も実行
理由：
現在の状態ですとifの条件がfalseであった場合redirect_to root_pathが読まれません。
destroyアクションが動いた後、遷移するviewが存在せず、意図通りの挙動にならないためです。

naoya1215 commented:
有難うございます。
確かにその通りですね。理解しました有難うございます。
修正しましたのでよろしくお願いします。

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます！

修正点を確認いたしました。

**LGTM**とさせていただきます！
削除機能の実装お疲れ様でした！
レビュー依頼のgoogleフォーマットのみならず、コメントも添えていただきありがとうございます。
ご理解いただけたことのご連絡もありがとうございました。
削除機能という何気ない機能ですが、いざ実装しようと思ったら細かいところにも注目しなくてはいけません。意外と難しいですよね。

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に
次の実装のためのブランチを切って次の課題に取り組みましょう！

引き続き、開発頑張っていきましょう！応援しております。

naoya1215 committed:
コメント追加
